lib/src/strings.dart

```diff
@@ -67,142 +54,220 @@ String _yamlEncodeDoubleQuoted(String string) {
-  /// `\r\n` already
-  return result +
-      trimmedString.replaceAll('\n', lineEnding * 2 + ' ' * indentation) +
-      removedPortion;
+/// See https://yaml.org/spec/1.2.2/#8112-block-chomping-indicator
+String _getChompingIndicator(String string) {
+  /// By default, we apply an indent to the string after every new line.
+  ///
+  /// Apply the `keep (+)` chomping indicator for trailing whitespace to be
+  /// treated as content.
+  ///
+  /// [NOTE]: We only check for trailing whitespace rather than `\n `. This is
+  /// a coin-toss approach. If there is a new line after this, it will be kept.
+  /// If not, nothing happens.
+  if (string.endsWith(' ') || string.endsWith('\n')) return '+';
+
+  return '-';
 }
 
-/// Generates a YAML-safe literal string.
+/// Attempts to encode a [string] as a _YAML folded string_ and apply the
+/// appropriate _chomping indicator_.
 ///
-/// It is important that we ensure that [string] is free of unprintable
-/// characters by calling [_hasUnprintableCharacters] before invoking this
-/// function.
-String _tryYamlEncodeLiteral(
-    String string, int indentation, String lineEnding) {
-  final result = '|-\n$string';
-
-  /// Assumes the user did not try to account for windows documents by using
-  /// `\r\n` already
-  return result.replaceAll('\n', lineEnding + ' ' * indentation);
-}
-
-/// Returns [value] with the necessary formatting applied in a flow context
-/// if possible.
+/// Returns `null`, if the [string] cannot be encoded as a _YAML folded
+/// string_.
 ///
-/// If [value] is a [YamlScalar], we try to respect its [YamlScalar.style]
-/// parameter where possible. Certain cases make this impossible (e.g. a plain
-/// string scalar that starts with '>'), in which case we will produce [value]
-/// with default styling options.
-String _yamlEncodeFlowScalar(YamlNode value) {
-  if (value is YamlScalar) {
-    assertValidScalar(value.value);
-
-    final val = value.value;
-    if (val is String) {
-      if (_hasUnprintableCharacters(val) ||
-          value.style == ScalarStyle.DOUBLE_QUOTED) {
-        return _yamlEncodeDoubleQuoted(val);
-      }
+/// **Examples** of folded strings.
+/// ```yaml
+/// # With the "strip" chomping indicator
+/// key: >-
+///   my folded
+///   string
+///
+/// # With the "keep" chomping indicator
+/// key: >+
+///   my folded
+///   string
+/// ```
+///
+/// See: https://yaml.org/spec/1.2.2/#813-folded-style
+String? _tryYamlEncodeFolded(String string, int indentSize, String lineEnding) {
+  if (_hasUnprintableCharacters(string)) return null;
+  // A string that starts with space or newline followed by space can't be
+  // encoded in folded mode.
+  if (string.startsWith(' ') && string.startsWith('\n ')) return null;
 
-      if (value.style == ScalarStyle.SINGLE_QUOTED) {
-        return _tryYamlEncodeSingleQuoted(val);
-      }
+  // TODO: Are there other strings we can't encode in folded mode?
```

Comment:
There must be other things we can't represent in folded mode, right?

Reply:
The leading space was the main culprit. 

Also the `if` statement for the leading space and line break (in view here) makes the tests fail 😅 I will carry one from your commit though.